django_tex/filters.py

```diff
@@ -1,20 +1,46 @@
 from django.utils.formats import localize_input
 from django.template.defaultfilters import register
 
+BACKSLASH_REPLACEMENT = "<:-!~#BACKSLASH#~!:->"
 
-def do_linebreaks(value):
+
+def do_linebreaks(value: str) -> str:
     return value.replace("\n", "\\\\\n")
 
 
-def do_latex_escape(value):
+def do_latex_escape(value: str) -> str:
+    """
+    Replace all LaTeX characters that could cause the latex compiler to fail
+    and at the same time try to display the character as intended from the user.
+
+    see also https://tex.stackexchange.com/questions/34580/escape-character-in-latex
+    """
+
     return (
-        value.replace("&", "\\&")
-        .replace("$", "\\$")
-        .replace("%", "\\%")
-        .replace("#", "\\#")
-        .replace("_", "\\_")
-        .replace("{", "\\{")
-        .replace("}", "\\}")
+        # Replace Backslashes so we dont' double replace backslashes
+        value.replace("\\", BACKSLASH_REPLACEMENT)
```

Comment:
Why replace `\\` with `BACKSLASH_REPLACEMENT`? I would say, if the input string contains a `\\`, the user wants it to be replaced with `\textbackslash\textbackslash` and not with a single `\textbackslash`.

Reply:
Simply put `print("\\")` in your python console to se what is happening.

Having "\\" is the only way to define a __single__ backslash.

Reply:
Oh, you're right. I was confused by the comment about not double replacing backslashes. I think, the same can be achieved using a regex subsitution as in commit 104163e below.